stellargraph/utils/saliency_maps/integrated_gradients.py

```diff
@@ -28,148 +28,208 @@
 """
 
 import numpy as np
-from .saliency import GradientSaliency
+from tensorflow.keras import backend as K
 from scipy.sparse import csr_matrix
+import tensorflow as tf
+from stellargraph.mapper import SparseFullBatchSequence, FullBatchSequence
 
 
-class IntegratedGradients(GradientSaliency):
+class IntegratedGradients:
     """
     A SaliencyMask class that implements the integrated gradients method.
     """
 
     def __init__(self, model, generator):
-        super().__init__(model, generator)
+        """
+        Args:
+            model (Keras model object): The differentiable graph model object.
+                For a dense model, the model.input should contain two tensors:
+                    - features: The placeholder of the feature matrix.
+                    - adj: The placeholder of the adjacency matrix.
+                For a sparse model, the model.input should contain three tensors:
+                    - features: The placeholder of the feature matrix.
+                    - adj_index: The placeholder of the adjacency matrix.
+                    - adj_values: The placeholder of the adjacency matrix.
+                The model.output (Keras tensor) is the tensor of model prediction output.
+                    This is typically the logit or softmax output.
+
+        """
+        # Set sparse flag from the generator
+        self.is_sparse = generator.use_sparse
```

Comment:
I think pretty much all the member variables defined here could be marked as private with a leading underscore? The user does end up constructing/interacting with this class if they want to do interpretability so it would save them from seeing a whole bunch of internal variables
```python
self._is_sparse
self._adj
self._adj_inds
self._features
self._model
```